srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Components/UnstuckMenu/UnstuckView.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace ACore\Components\UnstuckMenu;
+
+
+class UnstuckView
+{
+    private $controller;
+
+    public function __construct($controller)
+    {
+        $this->controller = $controller;
+    }
+
+    public function getUnstuckmenuRender($chars)
+    {
+        ob_start();
+
+        wp_enqueue_style('bootstrap-css', '//cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css', array(), '5.1.3');
+        wp_enqueue_style('acore-css', ACORE_URL_PLG . 'web/assets/css/main.css', array(), '0.1');
+        wp_enqueue_script('bootstrap-js', '//cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js', array(), '5.1.3');
+
+        // Enqueue a dummy script for localization
+        wp_enqueue_script('dummy-script', plugins_url('dummy.js', __FILE__), [], null, true);
+        // Localize script to pass REST API nonce and URL
+        wp_localize_script('dummy-script', 'myData', [
+            'nonce' => wp_create_nonce('wp_rest'), // Pass REST API nonce
+            'restUrl' => get_rest_url(null, ACORE_SLUG . '/v1/unstuck') // Pass the REST URL
+        ]);
+
+        // Add inline script with the localization data
+        wp_add_inline_script('dummy-script', '
+            jQuery(document).ready(function() {
+                jQuery(".unstuck-button").on("click", function() {
+                    const charName = jQuery(this).data("char-name");
+                    jQuery.ajax({
+                        type: "POST",
+                        url: myData.restUrl, // Use the localized REST URL
+                        data: {
+                            charName: charName
+                        },
+                        headers: {
+                            "X-WP-Nonce": myData.nonce // Add the nonce to the request headers
+                        },
+                        xhrFields: {
+                            withCredentials: true // Ensure cookies are sent
+                        },
+                        success: function(response) {
+                            console.log(response);
+                            location.reload();
+                        },
+                        error: function(xhr, status, error) {
+                        }
+                    });
+                });
+            });
+        ');
+
+?>
+
+        <div class="wrap">
+            <div class="col-sm-4">
+                <div class="card">
+                    <div class="card-body">
+                        <h3>Unstuck</h3>
+                        <p>Unstuck your characters and teleport them to the hearthstone location</p>
+                        <hr>
+                        <ul id="acore-characters-unstuck" class="list-unstyled">
+                            <?php foreach ($chars as $char) {
+                                $currentTime = time();
+                                $isDisabled = ($char["time"] > $currentTime);
```

Comment:
Encountered edge-case: the buttons are never disabled when user used a day before the HS, but the DB doesn't get immediately updated.

1. use hs, gets cd, state is not yet persisted in DB
2. logout, state is persisted in DB
3. wait 1h, CD should be down
4. login, DB still has the CD even if it's in the past, but the client is allowing the player to use hs
5. logout (the CD is removed from DB)